app/src/app/[lng]/cities/[cityId]/HIAP/HiapTab/index.tsx

```diff
@@ -138,6 +138,34 @@ export function HiapTab({
     }
   }, [actions]);
 
+  const handleRowSelectionChange = async (
+    updaterOrValue:
+      | RowSelectionState
+      | ((old: RowSelectionState) => RowSelectionState),
+  ) => {
+    if (!inventory) return;
+
+    const newRowSelection =
+      typeof updaterOrValue === "function"
+        ? updaterOrValue(rowSelection)
+        : updaterOrValue;
+
+    try {
+      const selectedActionIds = Object.keys(newRowSelection).filter(
+        (id) => newRowSelection[id],
+      );
+      await updateHiapSelection({
+        inventoryId: inventory.inventoryId,
+        selectedActionIds,
+      }).unwrap();
+
+      setRowSelection(newRowSelection);
+      logger.info(selectedActionIds, "Updated selection");
+    } catch (error) {
+      logger.error(error, "Failed to update selection");
```

Comment:
We need a toast notification here, in case this fails. Ideally with a helpful message that has i18n and is more detailed than "something went wrong" - maybe by returning an error code from the backend and translating it here before creating a toast message.